docs/DeveloperGuide.adoc

```diff
@@ -153,94 +181,369 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+reduces the amount of switch cases within `DukeCooksParser` and makes testing more manageable.
 
-image::UndoRedoState1.png[]
+// tag::dashboard[]
+=== Dashboard features
+==== Implementation
+The functions add/delete/edit/find/list task, are found under the Dashboard. +
+The Dashboard in `DukeCooks` serves as a homepage which displays a list of tasks set out
+by the user. The tasks are sorted accordingly by earliest date.
+Here, a task is represented by a Dashboard object.
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+The add/delete/edit/find/list mechanism is implemented in `DukeCooks v1.3`. It extends
+`DukeCooks` with a function to add/delete/edit/find tasks.
 
-image::UndoRedoState2.png[]
+The operations are as followed:
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+* `DukeCooks v1.3#add()` -- Adds and saves the task into `DukeCooks`.
+* `DukeCooks v1.3#delete()` -- Removes the specified task from `DukeCooks`.
+* `DukeCooks v1.3k#edit()` -- Edits the specified with a new task name and/or date.
+* `DukeCooks v1.3#find()` -- Finds a list of task with a given keyword.
+* `DukeCooks v1.3#list()` -- Finds a list of task with a given keyword.
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+These operations are exposed in the Model interface as `Model#addDashboard()`,
+`Model#deleteDashboard()`, `Model#setDashboard()`, `Model#hasDashboard()` and
+`Model#getDashboardRecords()` respectively.
 
-image::UndoRedoState3.png[]
+Given below is an example usage scenario and how the add/delete/edit/find/list
+mechanism behaves at each step.
 
-[NOTE]
-If the `currentStatePointer` is at index 0, pointing to the initial address book state, then there are no previous address book states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the undo.
+Step 1. The user launches the application for the first time. `DukeCooks v1.3` will
+be initialized with the dashboard state, and the `currentStatePointer` pointing to
+that single dashboard state.
 
-The following sequence diagram shows how the undo operation works:
+Step 2. The user executes `add task tn/bake a cake td/12/12/2019` command to add a new
+task into Duke Cooks. The `add` command calls `Model#addDashboard()`, causing the task to
```

Comment:
might want to consider splitting this into smaller steps because it looks quite cluttered to the reader in my opinion